confdb/aclocal_cc.m4

```diff
@@ -1597,3 +1597,20 @@ AC_DEFUN([PAC_ARG_ATOMIC_PRIMITIVES], [
         lock - Mutex-based synchronization
         no|none - atomic operations are performed without synchronization
      ],,with_mpl_atomic_primitives=auto)])
+
+dnl
+dnl Check whether C compiler accepts -fno-common, if yes, add to CFLAGS.
+dnl
+dnl NOTE: standard C does not need common symbols. Use of common symbols
+dnl may bring issues on some systems, e.g. macos building static libraries.
+dnl As an anecodote, gcc starting with gcc-10 default to -fno-common.
+dnl
+dnl In the past we have added unnecessary initializers to global variables
+dnl to force not to generate common symbols. This macro obviates those fixes.
+dnl
+AC_DEFUN([PAC_C_NO_COMMON],[
+    # TODO: check if -fno-common is the default and skip
```

Comment:
We could simply add a `case $host in *-darwin*)`. What do you think, @raffenet ?

Reply:
Limiting the change to darwin sounds like a good idea. I was also digging around for info on this, and this thread has another solution, which is adding `-c` to the ranlib command to ensure the common symbols are included in the archive table of contents: https://lists.apple.com/archives/darwin-dev/2009/Mar/msg00137.html. Unfortunately the thread ended without concluding if this was a good idea or not. The help text quoted does seem to suggest it is not preferred. I confirmed in testing that it does solve the issue. You can either patch libtool.m4, or just set `RANLIB="ranlib -c"` at configure time, but again, it might not be preferred.

```
-c     Include common symbols as definitions with respect to the table of contents.  This is seldom the
              intended behavior for linking from a library, as it forces the linking of a library member just
              because it uses an uninitialized global that is undefined at that point in the linking.  This
              option is included only because this was the original behavior of ranlib.  This option is not the
              default.
```

Reply:
Yeah, I was considering this solution. It is really libtool's job to do that. OTOH, since there is really no reason to create common symbols for C, in the end, I decided it is simpler and once for all to just pass `-fno-common`. Maintaining yet another libtool patch seemed burdensome and unreliable.

Besides, I peaked inside `libtool.m4` -- what a mess -- I really don't want anything to do with it 😆

Reply:
ref: https://github.com/pmodels/mpich/issues/2436#issuecomment-1021674070

Reply:
Every C library project will encounter this issue (looks like since the beginning of osx), I really wonder why libtool hasn't taken care of this already.

Reply:
Adding `-fno-common` is ok with me, but I think I prefer limiting to macOS in the end.